src/md5.c

```diff
@@ -1,21 +1,19 @@
-#include "md5.h"
 #include "arch.h"
+#include "md5.h"
 
 #if defined(MG_ENABLE_MD5) && MG_ENABLE_MD5
 
 static void mg_byte_reverse(unsigned char *buf, unsigned longs) {
-/* Forrest: MD5 expect LITTLE_ENDIAN, swap if BIG_ENDIAN */
-#if BYTE_ORDER == BIG_ENDIAN
-  do {
-    uint32_t t = (uint32_t) ((unsigned) buf[3] << 8 | buf[2]) << 16 |
-                 ((unsigned) buf[1] << 8 | buf[0]);
-    *(uint32_t *) buf = t;
-    buf += 4;
-  } while (--longs);
-#else
-  (void) buf;
-  (void) longs;
-#endif
+  if (MG_BIG_ENDIAN) {
```

Comment:
Wouldn't this be better to use as a normal `#if`, `#else` macro so that it's more clear that it's a compile time conditional?

Reply:
A compiler optimises that anyways - so it'll be a no-op on a little endian systems.